golem-api-grpc/proto/golem/component/v1/component_service.proto

```diff
@@ -64,6 +65,8 @@ message CreateComponentRequestHeader {
   golem.common.ProjectId projectId = 1;
   string componentName = 2;
   optional ComponentType componentType = 3;
+  // All files need to be uploaded to the blob storage before providing them here
+  repeated InitialComponentFile files = 4;
```

Comment:
It feel a bit wasteful to upload the files through grpc if we are already in the 'trusted' context of the application and have direct access to the blob_storage, so I went with this approach

Reply:
I did not think of the gRPC API as something that is only available for internal communication (but it is right that nothing external uses it at the moment).